crates/bevy_ecs/tests/system_param.rs

```diff
@@ -0,0 +1,42 @@
+use bevy_ecs::prelude::*;
+use bevy_ecs::system::SystemParam;
+
+#[allow(dead_code)]
+#[derive(Debug, Resource)]
+struct PlayerName(String);
+
+#[derive(Debug, Component)]
+struct Companion;
+
+#[allow(dead_code)]
+#[derive(Debug, SystemParam)]
+struct UsualLifetime<'w> {
+    player_name: Res<'w, PlayerName>,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, SystemParam)]
+struct UsualLifetimes<'w, 's> {
+    player_name: Res<'w, PlayerName>,
+    companions: Query<'w, 's, &'static Companion>,
+}
```

Comment:
Is cargo or CI even running this file?

We have been putting `SystemParam` compile tests in `bevy_ecs/src/system/system_param.rs`, at the bottom of the file. These should probably go there too.

Reply:
Done.